sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v2/script.sql

```diff
@@ -0,0 +1,376 @@
+MERGE INTO
+  `moz-fx-data-shared-prod.mozilla_org_derived.ga_sessions_v2` T
+  USING (
+    WITH device_properties_at_session_start_event AS (
+      --get all session starts, from any date
+      SELECT
+        a.user_pseudo_id AS ga_client_id,
+        CAST(e.value.int_value AS string) AS ga_session_id,
+        (
+          SELECT
+            `value`
+          FROM
+            UNNEST(event_params)
+          WHERE
+            key = 'ga_session_number'
+          LIMIT
+            1
+        ).int_value AS ga_session_number,
+        geo.country AS country,
+        geo.region AS region,
+        geo.city AS city,
+        collected_traffic_source.manual_campaign_id AS campaign_id,
+        collected_traffic_source.manual_campaign_name AS campaign,
+        collected_traffic_source.manual_source AS source,
+        collected_traffic_source.manual_medium AS medium,
+        collected_traffic_source.manual_term AS term,
+        collected_traffic_source.manual_content AS content,
+        collected_traffic_source.gclid AS gclid,
+        device.category AS device_category,
+        device.mobile_model_name AS mobile_device_model,
+        device.mobile_marketing_name AS mobile_device_string,
+        device.operating_system AS os,
+        device.operating_system_version AS os_version,
+        device.language AS `language`,
+        device.web_info.browser AS browser,
+        device.web_info.browser_version AS browser_version,
+        PARSE_DATE('%Y%m%d', event_date) AS session_date,
+        ROW_NUMBER() OVER (
+          PARTITION BY
+            a.user_pseudo_id,
+            e.value.int_value
+          ORDER BY
+            a.event_timestamp ASC
+        ) AS rnk
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_2*` a
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        e.key = 'ga_session_id'
+        AND e.value.int_value IS NOT NULL
+        AND a.event_name = 'session_start'
+      QUALIFY
+        rnk = 1
+    ),
+    --get all the page views and min/max event timestamp and whether there was a product download
+    event_aggregates AS (
+      SELECT
+        a.user_pseudo_id AS ga_client_id,
+        CAST(e.value.int_value AS string) AS ga_session_id,
+        COUNTIF(event_name = 'page_view') AS pageviews,
+        MIN(event_timestamp) AS min_event_timestamp,
+        MAX(event_timestamp) AS max_event_timestamp,
+        CAST(
+          MAX(
+            CASE
+              WHEN event_name IN (
+                  'firefox_download',
+                  'firefox_mobile_download',
+                  'focus_download',
+                  'klar_download'
+                )
+                THEN 1
+              ELSE 0
+            END
+          ) AS boolean
+        ) AS had_download_event
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_2*` a
```

Comment:
```suggestion
        `moz-fx-data-marketing-prod.analytics_313696158.events_2*` AS a
```

Could we also avoid cryptic, un-descriptive naming like `x`?

As per SQL style guide, we should aim aliases to be meaningful or descriptive.

Reply:
Fixed - thanks!